From 0b7961f55262dcb090c6284b88e60f9d162da31d Mon Sep 17 00:00:00 2001
From: sveinung <sveinung@a0f10bec-cc02-0410-94fc-a9cfff90b4cd>
Date: Wed, 12 Apr 2017 09:53:58 +0000
Subject: [PATCH 2/6] Use 3.0 devel sg compatibility while it is there.

Freeciv 3.1 still has 3.0 development version save game compatibility code.
Use it.

See hrm Feature #651394

git-svn-id: svn://svn.gna.org/svn/freeciv/trunk@35220 a0f10bec-cc02-0410-94fc-a9cfff90b4cd
---
 server/savecompat.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/server/savecompat.c b/server/savecompat.c
index c99346d..410f373 100644
--- a/server/savecompat.c
+++ b/server/savecompat.c
@@ -163,7 +163,12 @@ void sg_load_compat(struct loaddata *loading, enum sgf_version format_class)
   }
 
 #ifdef FREECIV_DEV_SAVE_COMPAT
-  if (loading->version == compat[compat_current].version) {
+  if (loading->version == compat[compat_current].version
+#ifdef FREECIV_DEV_SAVE_COMPAT_3_0
+      /* Make use of 3.0 compatibility while it is there. */
+      || loading->version == 40
+#endif /* FREECIV_DEV_SAVE_COMPAT_3_0 */
+      ) {
     compat_load_dev(loading);
   }
 #endif /* FREECIV_DEV_SAVE_COMPAT */
-- 
2.1.4

